hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -328,13 +406,16 @@ public AbfsRestOperation renamePath(String source, final String destination, fin
   }
 
   public AbfsRestOperation append(final String path, final long position, final byte[] buffer, final int offset,
-                                  final int length, boolean flush, boolean isClose)
+                                  final int length, boolean flush, boolean isClose, final String leaseId)
```

Comment:
this seems to be recurrent merge pain point: too many patches adding more things to every rest call. 

Proposed: how about adding a RestOperationContext struct which gets passed down, leaseId would go in there, and later other stuff (statistics, trace context, etc) ?